apps/functions-subscription/src/adapters/azure/cosmosdb/activation-job.ts

```diff
@@ -0,0 +1,45 @@
+import * as t from 'io-ts';
+import * as O from 'fp-ts/lib/Option';
+import * as TE from 'fp-ts/lib/TaskEither';
+import * as RA from 'fp-ts/lib/ReadonlyArray';
+import { flow, pipe } from 'fp-ts/lib/function';
+import { InvocationContext } from '@azure/functions';
+import {
+  ActivationJobItemCodec,
+  ActivationRequestItemCodec,
+} from '../../../domain/activation';
+import { SystemEnv } from '../../../system-env';
+import { Config } from '../../../config';
+
+export const makeActivationJobCosmosHandler =
+  (env: SystemEnv, { activations }: Config) =>
+  (
+    documents: unknown,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    context: InvocationContext,
+  ): Promise<unknown> =>
+    pipe(
+      // documents is an array of documents of the activations container
+      TE.fromEither(
+        t
+          .array(t.union([ActivationRequestItemCodec, ActivationJobItemCodec]))
+          .decode(documents),
+      ),
+      // Keep only job documents
+      TE.map(
+        flow(
+          RA.filterMap((doc) => (doc.type === 'job' ? O.some(doc) : O.none)),
+        ),
+      ),
+      TE.flatMap(([job]) =>
+        // Call the method to activate users only if the updated document is a job
+        job
+          ? env.processActivationJob(job, activations.concurrencyThreshold)
+          : TE.right(['success' as const]),
```

Comment:
```suggestion
          : TE.right(['not-executed' as const]),
```
maybe we can use the `not-executed` result